File: 1100/2065C1/main.java
import java.util.*;
 
public class main
{
    public static void main(String[]args)
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int m = sc.nextInt();
            
            int l = 0;
            
            int []a = new int[n];
            for(int i=0;i<n;i++)
                a[i] = sc.nextInt();
                
            int b = sc.nextInt();
            for(int i=0;i<n;i++)
            {
                a[i] = Math.min(a[i], b-a[i]);
                if(i !=0 && a[i] < a[i-1])
                    a[i] = b-a[i];
            }
            
            for(int i=0;i<n-1;i++)
            {
                if(a[i] <= a[i+1])
                    l++;
            }
        
            System.out.println(l==n-1?"YES":"NO");
        }
    }
}

File: 1100/2065C1/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n, m;
        cin >> n >> m;

        vector<int> a(n);
        for (int i = 0; i < n; i++) 
            cin >> a[i];

        int b;
        cin >> b;

        for (int i = 0; i < n; i++) 
        {
            a[i] = min(a[i], b - a[i]);
            if (i != 0 && a[i] < a[i - 1]) 
                a[i] = b - a[i];
        }

        int l = 0;
        for (int i = 0; i < n - 1; i++) 
        {
            if (a[i] <= a[i + 1]) 
                l++;
        }

        cout << (l == n - 1 ? "YES" : "NO") << "\n";
    }

    return 0;
}
